core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,224 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', '$filter', '$injector',
+  'editorContextService', 'alertsService', 'INTERACTION_SPECS',
+  'stateSolutionService', 'explorationStatesService',
+  'StateSolutionHelperService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'AnswerClassificationService',
+  'explorationContextService', 'angularNameService',
+  function(
+    $scope, $rootScope, $modal, $filter, $injector,
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    StateSolutionHelperService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.StateSolutionHelperService = StateSolutionHelperService;
+    $scope.inlineSolutionEditorIsActive = false;
+    $scope.interactionHtml = '';
+    $scope.SOLUTION_EDITOR_FOCUS_LABEL = (
+      'currentInteractionHtmlForSolutionEditor');
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.correctAnswer = null;
+
+    $scope.interactionHtml = (
+      oppiaExplorationHtmlFormatterService.getInteractionHtml(
+        stateInteractionIdService.savedMemento,
+        explorationStatesService.getInteractionCustomizationArgsMemento(
+          editorContextService.getActiveStateName()),
+        $scope.SOLUTION_EDITOR_FOCUS_LABEL));
+
+    $scope.isSolutionValid = function() {
+      return explorationStatesService.getState(
+        editorContextService.getActiveStateName()
+      ).interaction.isSolutionValid();
+    };
+
+    $scope.inlineSolutionEditorIsActive = false;
+    var interactionId = stateInteractionIdService.savedMemento;
+
+    if (!StateSolutionHelperService.isSupportedInteraction(interactionId)) {
+      // In this case the interactionHtml is constructed in the Solution
+      // object using the objectType. So interactionHtml is set to null here.
+      $scope.interactionHtml = null;
+    }
+    $scope.objectType = StateSolutionHelperService.getInteractionObjectType(
```

Comment:
The objectType and interactionHtml aren't entirely exclusive. The objectType is used to fetch the right summary in the minified solution editor. So even if a solution is not manually constructed, it might have an objectType var.